ambry-protocol/src/main/java/com.github.ambry.protocol/ReplicationControlAdminRequest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.protocol;
+
+import com.github.ambry.utils.Utils;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * An admin request used to control replication behavior (enable/disable)
+ */
+public class ReplicationControlAdminRequest extends AdminRequest {
```

Comment:
Shouldn't we also think of exposing admin request to check for current list of disabled nodes and partitions for a given storage node? Also, do add java docs to call out about persistence of these requests? What happens when storage nodes are restarted. May be a one page design doc would have been easier to go about rather than going through this during review